### PR TITLE
Improve cert signing integration test+:

### PR DESCRIPTION
**Pull request by bryanchriswhite**
+ improve bash utils/test
+ add verify command
+ reorganize certificates commands
+ reorganize certficate configs
+ refactor certificate backup

### CODE REVIEW

File: pkg/certificates/config.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+		err = errs.Combine(err, authDB.Close())
+	}()
+
+	signer, err := c.CA.Load()
+	if err != nil {
+		return err
+	}
+
+	srv := &CertificateSigner{
+		Log:           zap.L(),
```
Comment (mobyvb) [inline pkg/certificates/config.go]: does `Log` need to be public? In most other stuff I've seen it declared as lowercase.

File: scripts/test-certificate-signing.sh [hunk 2]
```diff
@@ -3,10 +3,10 @@ set -o errexit
```
Comment (mobyvb) [inline scripts/test-certificate-signing.sh]: is travis currently running this test?

File: cmd/certificates/auth.go [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+	authCmd.AddCommand(authCreateCmd)
+	cfgstruct.Bind(authCreateCmd.Flags(), &authCreateCfg, cfgstruct.ConfDir(defaultConfDir))
+	authCmd.AddCommand(authInfoCmd)
+	cfgstruct.Bind(authInfoCmd.Flags(), &authInfoCfg, cfgstruct.ConfDir(defaultConfDir))
+	authCmd.AddCommand(authExportCmd)
+	cfgstruct.Bind(authExportCmd.Flags(), &authExportCfg, cfgstruct.ConfDir(defaultConfDir))
+}
+
+func cmdCreateAuth(cmd *cobra.Command, args []string) error {
+	count, err := strconv.Atoi(args[0])
```
Comment (coyle) [inline cmd/certificates/auth.go]: is there any concern args will ever be empty?

Comment (stefanbenten) [reply cmd/certificates/auth.go]: He specified a minimum above in L34. That should prevent it from being empty

File: pkg/certificates/config.go [hunk 4]
```diff
@@ -0,0 +1,206 @@
+// Copyright (C) 2018 Storj Labs, Inc.
```
Comment (mniewrzal) [inline pkg/certificates/config.go]: 2019

File: pkg/certificates/config.go [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+		defer cancel()
+
+		ca, err = caConfig.Create(ctx)
+		if err != nil {
+			return err
+		}
+	}
+
+	identStatus := identConfig.Status()
+	if identStatus == identity.CertKey && !identConfig.Overwrite {
```
Comment (mniewrzal) [inline pkg/certificates/config.go]: nit - switch can be used here

File: cmd/certificates/setup.go [hunk 6]
```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2018 Storj Labs, Inc.
```
[the remark this answers is not in the document]
Comment (mniewrzal) [reply cmd/certificates/setup.go]: I believe that code created in 2018 needs to stay with date of creation.

File: cmd/certificates/verify.go [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+		RunE: cmdVerify,
+	}
+
+	verifyCfg verifyConfig
+)
+
+type checkOpts struct {
+	verifyConfig
+	ca       *identity.FullCertificateAuthority
+	ident    *identity.FullIdentity
```
[the remark this answers is not in the document]
Comment (bryanchriswhite) [reply cmd/certificates/verify.go]: I'm not convinced that's necessary.. I've consistently used `ident` for variables; this is an non-exported type and I think the `ident` is a decent compromise between clarity and conciseness.

[the remark this answers is not in the document]
Comment (egonelbre) [reply cmd/certificates/verify.go]: I would go with `identity` as well.

File: pkg/certificates/config.go (hunk 5 above)
Comment (bryanchriswhite) [reply pkg/certificates/config.go]: How? The conditions use logical and in and negation. I don't see a way to reduce this to a switch.

Comment (mniewrzal) [reply pkg/certificates/config.go]: An example https://github.com/storj/storj/blob/8893884044bcbe636c093e64df2931a5fd6b87b5/pkg/piecestore/psserver/server.go#L270

Comment (egonelbre) [reply pkg/certificates/config.go]:
```
switch {
case identStatus == identity.CertKey && !identConfig.Overwrite:
...
case identStatus != identity.NoCertNoKey && !identConfig.Overwrite:
...
default:
...
}

File: cmd/certificates/setup.go [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+	setupCmd = &cobra.Command{
+		Use:         "setup",
+		Short:       "Setup a certificate signing server",
+		RunE:        cmdSetup,
+		Annotations: map[string]string{"type": "setup"},
+	}
+
+	setupCfg struct {
+		// NB: cert and key paths overridden in setup
+		CA identity.CASetupConfig
```
[the remark this answers is not in the document]
Comment (bryanchriswhite) [reply cmd/certificates/setup.go]: Thanks for the info on `cfgstruct.BindSetup` and the `setup:"true"` tag, I was not aware of that. These fields are all used in `run` as well, the certificates service needs an identity as it is a grpc service.

**Pull request merged**